src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -1941,19 +2536,24 @@ public enum RisingRivals implements LogicCardInfo {
         return evolution (this, from:"Spheal", hp:HP080, type:WATER, retreatCost:2) {
           weakness M, PLUS20
           move "Find Ice", {
-            text "Energy cards, show them to your opponent, and put them into your hand. Shuffle your deck afterward."
-            energyCost W
-            attackRequirement {}
+            text "Search your deck for up to 2 [W] Energy cards, show them to your opponent, and put them into your hand. Shuffle your deck afterward."
+            energyCost ()
+            attackRequirement {
+              assert my.deck : "Your deck is empty"
+            }
             onAttack {
-              damage 0
+              my.deck.search(max:2,"Search your deck for up to 2 [W] Energy cards",energyFilter(W)).moveTo(my.hand)
```

Comment:
Shuffle that deck :laughing: 

Question for the future though, is there any reason that DeckCardList.search shouldn't Shuffle after each search? Maybe we could add that there if there aren't effects that would be affected by such a change.

Reply:
That's a really good idea!